### PR TITLE
[NIFI-13539] upgrade to angular 18.0.7

### PR DESCRIPTION
**Pull request by scottyaslan**

### CODE REVIEW

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Builds on all platforms are failing when invoking the `nx` command, so it appears something is missing in the configuration.

**Review by exceptionfactory — COMMENTED** [review]
Here is the particular error, which appears to be related to some conflicting dependency versions:

```
[INFO] > nifi-frontend@0.0.0 ci
[INFO] > npm ci --ignore-scripts
[INFO] 
[INFO] npm error code ERESOLVE
[INFO] npm error ERESOLVE could not resolve
[INFO] npm error
[INFO] npm error While resolving: @angular-eslint/schematics@17.2.1
[INFO] npm error Found: @angular/cli@18.0.7
[INFO] npm error node_modules/@angular/cli
[INFO] npm error   dev @angular/cli@"~18.0.0" from the root project
[INFO] npm error
[INFO] npm error Could not resolve dependency:
[INFO] npm error peer @angular/cli@">= 17.0.0 < 18.0.0" from @angular-eslint/schematics@17.2.1
[INFO] npm error node_modules/@angular-eslint/schematics
[INFO] npm error   dev @angular-eslint/schematics@"17.2.1" from the root project
[INFO] npm error
[INFO] npm error Conflicting peer dependency: @angular/cli@17.3.8
[INFO] npm error node_modules/@angular/cli
[INFO] npm error   peer @angular/cli@">= 17.0.0 < 18.0.0" from @angular-eslint/schematics@17.2.1
[INFO] npm error   node_modules/@angular-eslint/schematics
[INFO] npm error     dev @angular-eslint/schematics@"17.2.1" from the root project
[INFO] npm error
[INFO] npm error Fix the upstream dependency conflict, or retry
[INFO] npm error this command with --force or --legacy-peer-deps
[INFO] npm error to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by scottyaslan** [comment]
> Here is the particular error, which appears to be related to some conflicting dependency versions:
> 
> ```
> [INFO] > nifi-frontend@0.0.0 ci
> [INFO] > npm ci --ignore-scripts
> [INFO] 
> [INFO] npm error code ERESOLVE
> [INFO] npm error ERESOLVE could not resolve
> [INFO] npm error
> [INFO] npm error While resolving: @angular-eslint/schematics@17.2.1
> [INFO] npm error Found: @angular/cli@18.0.7
> [INFO] npm error node_modules/@angular/cli
> [INFO] npm error   dev @angular/cli@"~18.0.0" from the root project
> [INFO] npm error
> [INFO] npm error Could not resolve dependency:
> [INFO] npm error peer @angular/cli@">= 17.0.0 < 18.0.0" from @angular-eslint/schematics@17.2.1
> [INFO] npm error node_modules/@angular-eslint/schematics
> [INFO] npm error   dev @angular-eslint/schematics@"17.2.1" from the root project
> [INFO] npm error
> [INFO] npm error Conflicting peer dependency: @angular/cli@17.3.8
> [INFO] npm error node_modules/@angular/cli
> [INFO] npm error   peer @angular/cli@">= 17.0.0 < 18.0.0" from @angular-eslint/schematics@17.2.1
> [INFO] npm error   node_modules/@angular-eslint/schematics
> [INFO] npm error     dev @angular-eslint/schematics@"17.2.1" from the root project
> [INFO] npm error
> [INFO] npm error Fix the upstream dependency conflict, or retry
> [INFO] npm error this command with --force or --legacy-peer-deps
> [INFO] npm error to accept an incorrect (and potentially broken) dependency resolution.
> ```

Thanks @exceptionfactory! This was my first attempt at upgrading the dependencies of our Nx monorepository using their tools to do so and while I did find them helpful it didn't solve all the upgrade issues. I think I have them all sorted now but I will continue babysitting this build until it is green.

File: nifi-frontend/src/main/frontend/apps/nifi-jolt-transform-ui/src/app/app.module.ts [hunk 1]
```diff
@@ -55,8 +51,14 @@ import { JoltTransformJsonUiEffects } from './pages/jolt-transform-json-ui/state
     ],
     providers: [
         { provide: MAT_FORM_FIELD_DEFAULT_OPTIONS, useValue: { appearance: 'outline' } },
-        provideHttpClient(withInterceptors([]))
-    ],
-    bootstrap: [AppComponent]
+        provideHttpClient(withInterceptors([])),
+        provideHttpClient(
```
Comment (mcgilman) [inline nifi-frontend/src/main/frontend/apps/nifi-jolt-transform-ui/src/app/app.module.ts:55]: Do we need both `provideHttpClient` calls?

File: nifi-frontend/src/main/frontend/apps/nifi/src/app/app.module.ts [hunk 2]
```diff
@@ -96,8 +92,14 @@ import { BannerTextEffects } from './state/banner-text/banner-text.effects';
     ],
     providers: [
         { provide: MAT_FORM_FIELD_DEFAULT_OPTIONS, useValue: { appearance: 'outline' } },
-        provideHttpClient(withInterceptors([authInterceptor, loadingInterceptor, pollingInterceptor]))
-    ],
-    bootstrap: [AppComponent]
+        provideHttpClient(withInterceptors([authInterceptor, loadingInterceptor, pollingInterceptor])),
+        provideHttpClient(
+            withInterceptorsFromDi(),
```
Comment (mcgilman) [inline nifi-frontend/src/main/frontend/apps/nifi/src/app/app.module.ts:97]: I think we can be condensed. All of our interceptors have been converted to function based so I don't think we need the `widthInterceptorsFromDi`.

**Review by mcgilman — COMMENTED** [review]
Thanks @scottyaslan! Just a couple things noted below.

File: nifi-frontend/src/main/frontend/package.json [hunk 3]
```diff
@@ -44,46 +45,47 @@
...
     },
     "devDependencies": {
-        "@angular-builders/jest": "^17.0.2",
-        "@angular-devkit/build-angular": "17.2.0",
-        "@angular-devkit/core": "~17.1.3",
-        "@angular-devkit/schematics": "~17.1.3",
+        "@angular-builders/jest": "^18.0.0",
+        "@angular-devkit/build-angular": "18.0.7",
+        "@angular-devkit/core": "18.0.7",
         "@angular-eslint/builder": "17.2.1",
```
Comment (mcgilman) [inline nifi-frontend/src/main/frontend/package.json:51]: Is this dependency needed? If yes, should it have been updated too? It was the source of a npm installation warning.

**Review by mcgilman — APPROVED** [review]
Thanks @scottyaslan! Will merge once CI completes.

**Pull request merged**